susemanager-utils/susemanager-sls/src/states/uyuni_config.py

```diff
@@ -0,0 +1,757 @@
+            raise exc
+        else:
+            if __opts__['test']:
+                return StateResult.prepare_result(name, None, "{0} would be removed".format(name))
+            try:
+                __salt__['uyuni.systemgroup_delete'](name,
+                                                     org_admin_user=org_admin_user,
+                                                     org_admin_password=org_admin_password)
+                return StateResult.prepare_result(name, True, "Group {} has been deleted".format(name),
+                                                  {'name': {'old': current_group.get('name')},
+                                                   'description': {'old': current_group.get('description')}})
+            except Exception as exc:
+                return StateResult.state_error(name, "Error deleting group '{}': {}".format(name, exc))
+
+
+class UyuniOrgs:
+
+    @staticmethod
+    def _compute_changes(user_changes: Dict[str, Any],
+                         current_user: Dict[str, Any]) -> Dict[str, Any]:
+        changes = {}
+        for field in ["email", "first_name", "last_name"]:
+            if (current_user or {}).get(field) != user_changes.get(field):
+                changes[field] = {"new": user_changes[field]}
+                if current_user:
+                    changes[field]["old"] = (current_user or {}).get(field)
+        return changes
+
+    def manage(self, name: str, org_admin_user: str, org_admin_password: str, first_name: str,
+               last_name: str, email: str, pam: bool = False,
+               admin_user=None, admin_password=None) -> Dict[str, Any]:
+        """
+        Manage organization.
+        Admin user must have SUSE Manager Administrator role to perform this action
+
+        :param name: organization name
+        :param org_admin_user: organization admin user
+        :param org_admin_password: organization admin password
+        :param first_name: organization admin first name
+        :param last_name: organization admin last name
+        :param email: organization admin email
+        :param pam: organization admin pam authentication
+        :param admin_user: uyuni admin user
+        :param admin_password: uyuni admin password
+        :return: dict for Salt communication
+        """
+        current_org = None
+        current_org_admin = None
+        try:
+            current_org = __salt__['uyuni.org_get_details'](name,
+                                                            admin_user=admin_user,
+                                                            admin_password=admin_password)
+            current_org_admin = __salt__['uyuni.user_get_details'](org_admin_user,
+                                                                   org_admin_user=org_admin_user,
+                                                                   org_admin_password=org_admin_password)
+        except Exception as exc:
+            if exc.faultCode != 2850:
```

Comment:
I would suggest adding a comment to explain what this particular fault is about (or extract the constant with an intelligible name)

Reply:
I extract them to a constant. Those codes are error codes returned by XML-RPC API